### PR TITLE
Edit run - confirmation prompts

### PR DESCRIPTION
**Pull request by hjoliver**
This adds a post-edit confirmation prompt (CLI and GUI) to the new edit run feature, which allows users to back out if they think they have screwed up the edit (another one for BoM).

It also fixes a number of bugs identified while implementing this:
 * edit runs failed on tasks that hadn't run yet, and those that had run prior to a suite restart, because the job file path hadn't been computed in the task proxy yet.
 * if interactive command prompts were turned on in global config, the GUI prompts would always return False (abort the command).  Presumably no one uses interactive prompting
 * removed unused --notify-completion command options
 * a few command args were still not undergoing cycle point standardisation (i.e. conversion to the suite's cycle point format) in the daemon, so the command would fail if you used a different format

### CODE REVIEW

File: lib/cylc/scheduler.py [hunk 1]
```diff
@@ -1114,13 +1107,6 @@ def shutdown( self, reason='' ):
 
     def set_stop_point( self, stop_point_string ):
         stop_point = get_point(stop_point_string)
-        try:
-            stop_point.standardise()
-        except PointParsingError as exc:
-            self.log.critical(
-                "Cannot set stop cycle point: %s: %s" % (
-                    stop_point_string, exc))
-            return
```
Comment (hjoliver) [inline lib/cylc/scheduler.py]: Standardising is done in ```command_set_stop_after_point()```, which feeds into this.

**Comment by hjoliver** [comment]
@matthewrmshin - please review (or reassign).

Note that I've assigned this to next release as it contains a fairly important bug fix and it should be a pretty easy review.  But feel free to demote to soon if you haven't got the time today, in which case I'll post the new release immediately.

**Comment by hjoliver** [comment]
My test battery passes.

**Comment by arjclark** [comment]
@hjoliver - unfortunately Matt's now off for the Easter weekend, back Tuesday in theory but potentially away for school holidays.

**Comment by hjoliver** [comment]
@arjclark - ok, you wanna do it :grin: ?

**Comment by arjclark** [comment]
Just out my way out the door myself I'm afraid. You could try @kaday as she's in this afternoon or I can do it on tuesday.

**Comment by hjoliver** [comment]
Fair enough - have a nice break!  @kaday - what do you think?  No pressure, I suppose that everyone in the western world, including myself, is technically off work over easter, so we could just delay it all (bug fix reviews and next release) till next week...

**Comment by dpmatthews** [comment]
Delaying until next week makes sense to me.

**Comment by kaday** [comment]
I agree with Dave, I am having a quick look, but am not sure that I will get it done today.

File: lib/cylc/prompt.py [hunk 2]
```diff
@@ -17,14 +17,32 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import sys
+import gtk
```
Comment (matthewrmshin) [inline lib/cylc/prompt.py]: This line adds `gtk` to `cylc poll`, which is causing `tests/cylc-poll/03-poll-all.t` to fail in my environment.

I am getting a traceback like this in the `poll_now.20141207T0000Z` task of the test.

```
Traceback (most recent call last):
  File "/home/matt/cylc.git/bin/cylc-poll", line 26, in <module>
    from cylc.prompt import prompt
  File "/home/matt/cylc.git/lib/cylc/prompt.py", line 20, in <module>
    import gtk
  File "/usr/lib64/python2.6/site-packages/gtk-2.0/gtk/__init__.py", line 64, in <module>
    _init()
  File "/usr/lib64/python2.6/site-packages/gtk-2.0/gtk/__init__.py", line 52, in _init
    _gtk.init_check()
RuntimeError: could not open display
```

**Comment by matthewrmshin** [comment]
Functionality OK in an interactive environment. However, is it possible to display the task ID in the confirmation dialog?

**Comment by hjoliver** [comment]
Huh, the poll test passes in a non-interactive environment here.  Importing gtk just gives a warning, not a RuntimeError:
```
 python
Python 2.6.9 (unknown, Nov 19 2014, 15:44:49) 
[GCC 4.3.4 [gcc-4_3-branch revision 152973]] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import gtk
/usr/lib64/python2.6/site-packages/gtk-2.0/gtk/__init__.py:57: GtkWarning: could not open display
  warnings.warn(str(e), _gtk.Warning)
>>> 
```

**Comment by hjoliver** [comment]
I've moved the gtk import into the GUI code block.

**Comment by hjoliver** [comment]
Comments addressed.

File: lib/cylc/prompt.py [hunk 3]
```diff
@@ -19,12 +19,30 @@
 import sys
 
 from cylc.cfgspec.globalcfg import GLOBAL_CFG
+from cylc.gui.util import get_icon
```
Comment (matthewrmshin) [inline lib/cylc/prompt.py]: Need to remove this line as well. (`get_icon` is not used in this module, and it is causing another *cannot open display* failure.)

**Comment by arjclark** [comment]
@hjoliver - found a bug with manual testing if you click "no". Looks like the task submit number increments and the suite expects logs to be in place. So, if you make an edit, click no, subsequently click "View log files", you get a dialog with:

```
IOError: [Errno 2] No such file or directory: '/home/h06/aclark/cylc-run/test/log/job/1/foo/03/job.out'
```

**Comment by hjoliver** [comment]
@arjclark - the last commit writes "This edit run was aborted" to ```job.out``` and ```job.err``` in the job log dir, if the edit run is aborted.  Maybe it would be preferable to get the suite daemon to back off the task proxy's submit numbe, delete the new job log directory, and redo the ```NN``` symlink - as if the edit run was never initiated at all - but I think that is arguable as well as harder to achieve.

**Comment by arjclark** [comment]
@hjoliver - cool. I'm fine with the behaviour now as it has evidence of something happening.

**Comment by matthewrmshin** [comment]
Test battery OK in my environment.

However, I am now getting `00/job`, `00/job.out` and `00/job.err` in the dialog after an aborted edit run.

**Comment by matthewrmshin** [comment]
(In my tests, I have merged master into your branch. I don't know whether that causes a problem or not.)

**Comment by hjoliver** [comment]
In the prompt dialog, on a subsequent edit-run after aborting an earlier one?  I'm not getting that (tried merge from master too).  If you look at ```bin/cylc-trigger``` line 175, it's hard to see how that could happen?

**Comment by hjoliver** [comment]
I've identified one more problem with edit-run that would be good to have fixed on this branch: doing it to a waiting task that has not run yet results in a spurious '0' job submit number log directory and corresponding entries in the task proxy's log file list.  I can see why, but it's v late here so I'll bump this back to 'soon', and will release 6.4.0 in the morning.

**Comment by arjclark** [comment]
Just found another oddity.

 * run task
 * carry out an edit-run on a task but abort it
 * shutdown suite
 * restart suite
 * try to view log files for said task, getting dialog with "foo.1 has no associated log files"

My suite, for reference:

```
[scheduling]
[[dependencies]]
graph = foo:fail => bar

[runtime]
[[foo]]
    command scripting = sleep 10; echo $MESSAGE
    [[[environment]]]
        MESSAGE=hello
```

**Comment by arjclark** [comment]
Oddly, if I trigger and cancel another edit-run at that point I can then view log files for 03 and 02, but not 01...

**Comment by arjclark** [comment]
@matthewrmshin - when this is in, you may want to update rose bush as it shows the state of the task as being in "submit-init" for the following:

```
sqlite> select * from task_states where name is "foo";
foo|1|2015-04-08T14:33:51+01|2015-04-08T14:38:50+01|3||1|localhost|background|28343|succeeded
sqlite> select * from task_events where name is "foo";
foo|1|2015-04-08T14:33:51+01|1|incrementing submit number||localhost
foo|1|2015-04-08T14:33:52+01|1|submission succeeded||localhost
foo|1|2015-04-08T14:33:53+01|1|output completed|started|localhost
foo|1|2015-04-08T14:33:53+01|1|started||localhost
foo|1|2015-04-08T14:34:03+01|1|output completed|succeeded|localhost
foo|1|2015-04-08T14:34:03+01|1|succeeded||localhost
foo|1|2015-04-08T14:34:19+01|2|incrementing submit number||localhost
foo|1|2015-04-08T14:38:50+01|3|incrementing submit number||localhost
```

**Comment by hjoliver** [comment]
> Just found another oddity.

@arjclark - this is just a long-standing normal "feature" of restarts, as info about log files is not stored in the suite start dump (it should be!) ... i.e. nothing to do with this branch.  Or are you suggesting it's more than that?

**Comment by arjclark** [comment]
> @arjclark - this is just a long-standing normal "feature" of restarts, as info about log files is not stored in the suite start dump (it should be!) ... i.e. nothing to do with this branch. Or are you suggesting it's more than that?

Ah right, just never encountered it before then!

**Comment by hjoliver** [comment]
(this should now wait on #1503)

**Comment by hjoliver** [comment]
@arjclark and @matthewrmshin, this is back from the dead - please review.  I've fixed the issues mentioned above (not: https://github.com/cylc/cylc/pull/1401#issuecomment-91056516, which is not specific to this branch) and checked that submit number and gcylc "latest message" do the right thing for aborted edit runs, even for a 'waiting' task that has never run yet.    This could go in the next release if a timely review is feasible...

**Comment by matthewrmshin** [comment]
@hjoliver I cannot get these tests to pass in my environment:

```
./tests/jobscript/06-slurm-no-directives.t (Wstat: 0 Tests: 5 Failed: 3)
  Failed tests:  3-5
./tests/jobscript/07-traps-failure.t      (Wstat: 0 Tests: 13 Failed: 6)
  Failed tests:  3, 5, 7, 9, 11, 13
./tests/jobscript/00-torture.t            (Wstat: 0 Tests: 4 Failed: 1)
  Failed test:  4
./tests/jobscript/02-cycling.t            (Wstat: 0 Tests: 4 Failed: 1)
  Failed test:  4
./tests/env-filter/00-filter.t            (Wstat: 0 Tests: 17 Failed: 4)
  Failed tests:  14-17
./tests/jobscript/01-multi.t              (Wstat: 0 Tests: 22 Failed: 11)
  Failed tests:  4, 6-8, 10, 12, 14, 16, 18, 20, 22
./tests/hold-release/15-hold-after.t      (Wstat: 0 Tests: 2 Failed: 1)
  Failed test:  2
```

**Comment by hjoliver** [comment]
Hhmm, I should know by now, you can't beat the test battery :grimacing:

**Comment by hjoliver** [comment]
Fixed.

**Comment by matthewrmshin** [comment]
I can finally get the test battery to pass!

**Comment by matthewrmshin** [comment]
Functionality tested as working.

**Comment by matthewrmshin** [comment]
@arjclark or @kaday please review 2.

**Comment by arjclark** [comment]
@kaday - I'll take this one.

**Comment by arjclark** [comment]
Clicking on a task and trying to change its state, I get the following error dialog:

```
gcylc has a problem.

Traceback (most recent call last):
  File "/net/home/h06/aclark/cylc.git/lib/cylc/gui/app_gcylc.py", line 1709, in reset_task_state
    if not self.get_confirmation(cmd, task_id, msg):
TypeError: get_confirmation() takes exactly 2 arguments (4 given)
```

**Comment by hjoliver** [comment]
@arjclark - good spotting, bug fixed.

**Comment by arjclark** [comment]
Looks good to me. Tested as working in my environment.

**Pull request merged**